lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -240,21 +263,94 @@ data class PendingBlockWithTransactionsResponse(
 
     @SerialName("starknet_version")
     override val starknetVersion: String,
-) : GetBlockWithTransactionsResponse()
+) : GetBlockWithTransactionsResponse(), PendingBlock
+
+@Serializable
+sealed class GetBlockWithTransactionsReceiptsResponse : GetBlockWithResponse {
```

Comment:
I think we should get the naming in line, cause these are barely readable. I suggest the `TransactionReceipt` approach:

`GetBlockWithTransaction[...]Response` -> `BlockWithTransaction[...]`
`BlockWithTransaction[...]Response` -> `ProcessedBlockWithTransaction[...]`
`PendingBlockWithTransaction[...]Response` -> `PendingBlockWithTransaction[...]`


where `[...] = [s/Hashes/Receipts]`

This is breaking, but I think it will make interaction with these easier in the future.

WDYT?